online-dashboard-app/src/Components/User/JobsTable.jsx

```diff
@@ -1,60 +1,115 @@
+  }, []);
+
   const handlePageChange = (pageNumber) => {
     setCurrentPage(pageNumber);
   };
+
+  // Get the page numbers to display based on currentPage and pageNumbersToShow
+  const getPageNumbers = () => {
+    const pageNumbers = [];
+    const maxPageButtons = pageNumbersToShow;
+
+    // Calculate the range of pages to display
+    let startPage = Math.max(1, currentPage - Math.floor(maxPageButtons / 2));
+    let endPage = Math.min(totalPages, startPage + maxPageButtons - 1);
+
+    // Adjust startPage if there aren't enough pages to show
+    if (endPage - startPage + 1 < maxPageButtons) {
+      startPage = Math.max(1, endPage - maxPageButtons + 1);
+    }
+
+    for (let i = startPage; i <= endPage; i++) {
+      pageNumbers.push(i);
+    }
+
+    return pageNumbers;
+  };
+
   return (
-    <div className={`bg-white shadow-md rounded-lg overflow-hidden ${className}`}>
-      <table className="min-w-full table-auto">
-        <thead>
-          <tr className="bg-indigo-100 text-indigo-800 font-semibold">
-            <th className="p-3 text-left">Company Name</th>
-            <th className="p-3 text-left">Role</th>
-            <th className="p-3 text-left">Batch</th>
-            <th className="p-3 text-left">Qualification</th>
-            <th className="p-3 text-left">Apply Link</th>
-          </tr>
-        </thead>
-        <tbody>
-          {currentJobs.map((job) => (
-            <tr key={job.id} className="border-t hover:bg-gray-50 transition">
-              <td className="p-3">{job.company_name}</td>
-              <td className="p-3">{job.role}</td>
-              <td className="p-3">{job.batch}</td>
-              <td className="p-3">{job.qualification}</td>
-              <td className="p-3">
-
-                <button
-                  className="bg-blue-500 text-white py-2 px-4 rounded-md hover:bg-blue-600"
-                >
-                  <a
-                    href={job.apply_link}
-                    target="_blank"
-                    rel="noopener noreferrer"
-                    className="text-black-600 hover:underline red"
-                  >
-                    Apply
-                  </a>
-                </button>
-              </td>
+    <div className={`bg-white shadow-md rounded-lg ${className}`}>
+      {/* Table for larger screens */}
+      <div className="hidden md:block">
+        <table className="w-full table-auto">
+          <thead>
+            <tr className="bg-indigo-100 text-indigo-800 font-semibold">
+              <th className="p-3 text-left">Company Name</th>
+              <th className="p-3 text-left">Role</th>
+              <th className="p-3 text-left">Batch</th>
+              <th className="p-3 text-left">Qualification</th>
+              <th className="p-3 text-left">Apply Link</th>
             </tr>
+          </thead>
+          <tbody>
+            {currentJobs.map((job) => (
+              <tr key={job.id} className="border-t hover:bg-gray-50 transition">

```

Comment:
This is good!
But we actually need to create a seperate component for the below div and call the component here by passing the values